File: mobile/src/types/recipe.ts
export interface Ingredient {
  name: string
  quantity: string
  group: string
}

export interface RecipeStep {
  order: number
  instruction: string
  duration_minutes?: number | null
  ingredients_used: string[]
  equipment: string[]
  image_url?: string | null
  clip_url?: string | null
}

export interface ParsedRecipe {
  title: string
  servings?: number | null
  prep_time_minutes?: number | null
  cook_time_minutes?: number | null
  calories_per_serving?: number | null
  allergens?: string[]
  ingredients: Ingredient[]
  steps: RecipeStep[]
}

export interface ParseRecipeResponse {
  recipe: ParsedRecipe
  used_ai: boolean
  step_image_notes?: string[]
}

export interface RecipeSummary {
  id: string
  title: string
  step_count: number
  used_ai: boolean
  created_at: string
}

export interface RecipeListResponse {
  items: RecipeSummary[]
  total: number
}

export interface RecipeDetailResponse {
  id: string
  title: string
  raw_text: string
  source_type: string
  source_url?: string | null
  used_ai: boolean
  recipe: ParsedRecipe
  created_at: string
}

export interface ReviewDraft {
  rawText: string
  recipe: ParsedRecipe
  usedAi: boolean
  sourceType?: string
  sourceUrl?: string
  stepImageNotes?: string[]
}

File: backend/app/schemas/recipe.py
from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel, Field


class Ingredient(BaseModel):
    name: str
    quantity: str = ""
    group: str = "Main"


class RecipeStep(BaseModel):
    order: int
    instruction: str
    duration_minutes: Optional[int] = None
    ingredients_used: List[str] = Field(default_factory=list)
    equipment: List[str] = Field(default_factory=list)
    image_url: Optional[str] = None
    clip_url: Optional[str] = None


class ParsedRecipe(BaseModel):
    title: str
    servings: Optional[int] = None
    prep_time_minutes: Optional[int] = None
    cook_time_minutes: Optional[int] = None
    calories_per_serving: Optional[int] = None
    allergens: List[str] = Field(default_factory=list)
    ingredients: List[Ingredient]
    steps: List[RecipeStep]


class ParseRecipeRequest(BaseModel):
    raw_text: str = Field(min_length=10, max_length=50000)
    source_url: Optional[str] = Field(default=None, max_length=2048)
    video_duration: Optional[float] = Field(default=None, ge=0)


class ParseRecipeResponse(BaseModel):
    recipe: ParsedRecipe
    used_ai: bool
    step_image_notes: List[str] = Field(default_factory=list)
